src/python/Utils/Utilities.py

```diff
@@ -10,6 +10,23 @@
 import sys
 from types import ModuleType, FunctionType
 from gc import get_referents
+from functools import reduce
+
+
+def reduceReport(reportList, expectedValue='OK'):
+    """
+    Simple function to aggregate a list of values (possibly outcomes of multiple
+    calls to couchDB or similar) to a single value.
+    :param report:        A list with accumulated report values
+    :param exepctedValue: The value with which the initial report list is expected
+                          to be filled. (any type) Default: 'OK'
+    :return:              Either the expected value (could be of any type) in the
+                          case when all the entries in the list are identical or
+                          False in the case when any of them deviates from the expected value.
+    """
+    if reduce(lambda x, y: x == y == expectedValue and expectedValue, reportList, expectedValue):
```

Comment:
As already expressed in my previous comments, I don't think we actually need this function. In addition, this expression looks cumbersome and it is hard to say what exactly it does. Or it could be just me.
Nonetheless, let us keep discussing/reviewing this PR before taking any action here.

Reply:
It does  a simple thing:
* Takes a list of values (`reportList`) and applies a two parameterized lambda expression to each of them sequentially aggregating the previous result as an argument to the next step - IIUC this should be  the standard  `reduce` behavior: https://docs.python.org/3/library/functools.html#functools.reduce, but more or less this is the 'key' expression for understanding it: `((((1+2)+3)+4)+5)` it tells it all. 
* The lambda expression on its own does the following: It simply compares if the tow arguments which it gets during the call are equal between each other and if they are equal if they match the expected value on top of that and if all of this is true returns the expected value otherwice the result of the comparission (which should be simply `False` 
* For initialization the sequence of steps in the `reduce` call an initial value is provided, which is the last argument - `expectedValue`, such that all comparisons are done  against the expected value - even starting from the very first one

Reply:
BTW this function is agnostic to the list provided... so it is applicable in many other cases as the so provided unittests shows it. I called it `reduceReport` only because of the context  under which I initially suggested it before we moved it here. If I need to be honest the best name for this function should be `reduceList` or something in that line.